<launch>

  <arg name="os1_hostname" default="" doc="hostname or IP in dotted decimal form of the sensor"/>
  <arg name="os1_udp_dest" default="" doc="hostname or IP where the sensor will send data packets"/>
  <arg name="os1_lidar_port" default="0" doc="port to which the sensor should send lidar data"/>
  <arg name="os1_imu_port" default="0" doc="port to which the sensor should send imu data"/>
  <arg name="replay" default="false" doc="do not connect to a sensor; expect /os1_node/{lidar,imu}_packets from replay"/>
  <arg name="lidar_mode" default="" doc="resolution and rate: either 512x10, 512x20, 1024x10, 1024x20, or 2048x10"/>
  <arg name="metadata" default="" doc="override default metadata file for replays"/>
  <arg name="viz" default="false" doc="whether to run a simple visualizer"/>
  <arg name="image" default="false" doc="publish range/intensity/noise image topic"/>

  <node pkg="ouster_ros" name="os1_node" type="os1_node" output="screen" required="true">
    <param name="~/lidar_mode" type="string" value="$(arg lidar_mode)"/>
    <param name="~/replay" value="$(arg replay)"/>
    <param name="~/os1_hostname" value="$(arg os1_hostname)"/>
    <param name="~/os1_udp_dest" value="$(arg os1_udp_dest)"/>
    <param name="~/os1_lidar_port" value="$(arg os1_lidar_port)"/>
    <param name="~/os1_imu_port" value="$(arg os1_imu_port)"/>
    <param name="~/metadata" value="$(arg metadata)"/>
  </node>

  <node pkg="ouster_ros" type="os1_cloud_node" name="os1_cloud_node" output="screen" required="true">
    <remap from="~/os1_config" to="/os1_node/os1_config"/>
    <remap from="~/lidar_packets" to="/os1_node/lidar_packets"/>
    <remap from="~/imu_packets" to="/os1_node/imu_packets"/>
  </node>

  <node if="$(arg viz)" pkg="ouster_ros" name="viz_node" type="viz_node" output="screen" required="true">
    <remap from="~/os1_config" to="/os1_node/os1_config"/>
    <remap from="~/points" to="/os1_cloud_node/points"/>
    <rosparam param="/viz_node/range_radii">[5, 10, 15, 20, 40, 80]</rosparam>
  </node>

  <node if="$(arg image)" pkg="ouster_ros" name="img_node" type="img_node" output="screen" required="true">
    <remap from="~/os1_config" to="/os1_node/os1_config"/>
    <remap from="~/points" to="/os1_cloud_node/points"/>
  </node>

</launch>
